http-client/src/main/java/no/nav/familie/http/interceptor/BearerTokenClientInterceptor.kt

```diff
@@ -36,37 +66,44 @@ class BearerTokenWithSTSFallbackClientInterceptor(private val oAuth2AccessTokenS
         if (erSystembruker()) {
             request.headers.setBearerAuth(stsRestClient.systemOIDCToken)
         } else {
-            request.headers.setBearerAuth(genererAccessToken(request, clientConfigurationProperties, oAuth2AccessTokenService))
+            request.headers.setBearerAuth(genererAccessToken(request,
+                                                             clientConfigurationProperties,
+                                                             oAuth2AccessTokenService,
+                                                             clientCredentialOrJwtBearer()))
         }
         return execution.execute(request, body)
     }
 }
 
 private fun genererAccessToken(request: HttpRequest,
                                clientConfigurationProperties: ClientConfigurationProperties,
-                               oAuth2AccessTokenService: OAuth2AccessTokenService) : String {
-    val clientProperties = clientPropertiesFor(request.uri, clientConfigurationProperties)
+                               oAuth2AccessTokenService: OAuth2AccessTokenService,
+                               grantType: OAuth2GrantType): String {
+    val clientProperties = clientPropertiesFor(request.uri,
+                                               clientConfigurationProperties,
+                                               grantType)
     val response: OAuth2AccessTokenResponse = oAuth2AccessTokenService.getAccessToken(clientProperties)
     return response.accessToken
 }
 
-private fun clientPropertiesFor(uri: URI, clientConfigurationProperties: ClientConfigurationProperties): ClientProperties {
-    val values = clientConfigurationProperties
-            .registration
-            .values
-            .filter { uri.toString().startsWith(it.resourceUrl.toString()) }
-    return if (values.size == 1) values.first() else filterForGrantType(values, uri)
```

Comment:
Noen tanker om fjerning av `(values.size == 1) values.first()`, kan kanskje være en breaking change?

Reply:
Hmm, virker greit tenker jeg. Er vel ingen apper som legger inn 2 configs for samme integrasjon? Og isåfall så ville jo uansett forrige logikk gjort det samme? 

Jeg tenker nesten dette er bedre, fordi om man wirer opp clientcred bønne og bare har obo config så vil det smelle før man treffer den eksterne tjenesten. (hvis jeg har forstått logikken riktig ;) )

Reply:
Tror både vi og dere har det.. 
Det denne eventuellt kan smelle på er hvis det er av typen `urn:ietf:params:oauth:grant-type:token-exchange` men vet ikke om noen bruker den?

Reply:
Ja, vi har flere configs for en integrasjon, men ikke med samme grant type?

Reply:
Tror ikke det er mulig, lurer på om token-support feiler då

Reply:
Da er det hvertfall ok 👍